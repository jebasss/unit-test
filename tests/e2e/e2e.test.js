const run = require("./run");
const server = require("./server");

describe("end-to-end integration", () => {
  beforeAll(() => server.start(3333));
  afterAll(() => server.stop());

  test("prints error and help message when no arguments given", async () => {
    const { stderr, stdout, exitCode } = await run();
    expect(exitCode).toBe(1);
    expect(stderr).toMatchSnapshot();
    expect(stdout).toEqual("");
  });

  test("prints help message when --help given", async () => {
    const { stderr, stdout, exitCode } = await run("--help");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output for a filename", async () => {
    const { stderr, stdout, exitCode } = await run("./test/sample.html");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output for multiple filenames", async () => {
    const { stderr, stdout, exitCode } = await run(
      "./test/sample.html",
      "./test/sample.html"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output for a URL", async () => {
    const { stderr, stdout, exitCode } = await run(
      "http://localhost:3333/sample.html"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output for a URL and filename", async () => {
    const { stderr, stdout, exitCode } = await run(
      "./test/sample.html",
      "http://localhost:3333/sample.html"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output to specified width for --width", async () => {
    const { stderr, stdout, exitCode } = await run(
      "--width",
      "40",
      "./test/sample.html"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output to specified width for -w", async () => {
    const { stderr, stdout, exitCode } = await run(
      "-w",
      "40",
      "./test/sample.html"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output to specified width for --indent", async () => {
    const { stderr, stdout, exitCode } = await run(
      "--indent",
      "4",
      "./test/sample.html"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints formatted output to specified width for -i", async () => {
    const { stderr, stdout, exitCode } = await run(
      "-i",
      "4",
      "./test/sample.html"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

});
