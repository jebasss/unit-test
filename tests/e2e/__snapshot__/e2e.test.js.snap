// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration prints error and help message when no arguments given 1`] = `
"purl <args> uri [...uri]

Options:
      --version  Show version number                                   [boolean]
  -w, --width    max width of final output                [number] [default: 80]
  -i, --indent   indent width for each line                [number] [default: 2]
      --help     Show help                                             [boolean]

Not enough non-option arguments: got 0, need at least 1"
`;

exports[`end-to-end integration prints formatted output for a URL 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet
      molestie vestibulum. Nunc malesuada lobortis mi et commodo. Aenean at diam
      non lectus porta laoreet. Curabitur in aliquam augue. Sed in maximus
      ligula. Suspendisse iaculis tincidunt varius. In sit amet nulla in erat
      consequat tempor. Praesent eget urna ornare, mattis magna et, interdum
      nisl. Proin non mi sed ligula mattis consectetur. Sed sagittis nisl mollis
      felis sollicitudin lobortis. Praesent rhoncus turpis id viverra gravida.
      Pellentesque consequat turpis ut ex commodo, in volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel cursus non, semper quis lectus. Fusce at
      justo vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
      mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>
"
`;

exports[`end-to-end integration prints formatted output for a URL and filename 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet
      molestie vestibulum. Nunc malesuada lobortis mi et commodo. Aenean at diam
      non lectus porta laoreet. Curabitur in aliquam augue. Sed in maximus
      ligula. Suspendisse iaculis tincidunt varius. In sit amet nulla in erat
      consequat tempor. Praesent eget urna ornare, mattis magna et, interdum
      nisl. Proin non mi sed ligula mattis consectetur. Sed sagittis nisl mollis
      felis sollicitudin lobortis. Praesent rhoncus turpis id viverra gravida.
      Pellentesque consequat turpis ut ex commodo, in volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel cursus non, semper quis lectus. Fusce at
      justo vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
      mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>

<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet
      molestie vestibulum. Nunc malesuada lobortis mi et commodo. Aenean at diam
      non lectus porta laoreet. Curabitur in aliquam augue. Sed in maximus
      ligula. Suspendisse iaculis tincidunt varius. In sit amet nulla in erat
      consequat tempor. Praesent eget urna ornare, mattis magna et, interdum
      nisl. Proin non mi sed ligula mattis consectetur. Sed sagittis nisl mollis
      felis sollicitudin lobortis. Praesent rhoncus turpis id viverra gravida.
      Pellentesque consequat turpis ut ex commodo, in volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel cursus non, semper quis lectus. Fusce at
      justo vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
      mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>
"
`;

exports[`end-to-end integration prints formatted output for a filename 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet
      molestie vestibulum. Nunc malesuada lobortis mi et commodo. Aenean at diam
      non lectus porta laoreet. Curabitur in aliquam augue. Sed in maximus
      ligula. Suspendisse iaculis tincidunt varius. In sit amet nulla in erat
      consequat tempor. Praesent eget urna ornare, mattis magna et, interdum
      nisl. Proin non mi sed ligula mattis consectetur. Sed sagittis nisl mollis
      felis sollicitudin lobortis. Praesent rhoncus turpis id viverra gravida.
      Pellentesque consequat turpis ut ex commodo, in volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel cursus non, semper quis lectus. Fusce at
      justo vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
      mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>
"
`;

exports[`end-to-end integration prints formatted output for multiple filenames 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet
      molestie vestibulum. Nunc malesuada lobortis mi et commodo. Aenean at diam
      non lectus porta laoreet. Curabitur in aliquam augue. Sed in maximus
      ligula. Suspendisse iaculis tincidunt varius. In sit amet nulla in erat
      consequat tempor. Praesent eget urna ornare, mattis magna et, interdum
      nisl. Proin non mi sed ligula mattis consectetur. Sed sagittis nisl mollis
      felis sollicitudin lobortis. Praesent rhoncus turpis id viverra gravida.
      Pellentesque consequat turpis ut ex commodo, in volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel cursus non, semper quis lectus. Fusce at
      justo vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
      mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>

<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet
      molestie vestibulum. Nunc malesuada lobortis mi et commodo. Aenean at diam
      non lectus porta laoreet. Curabitur in aliquam augue. Sed in maximus
      ligula. Suspendisse iaculis tincidunt varius. In sit amet nulla in erat
      consequat tempor. Praesent eget urna ornare, mattis magna et, interdum
      nisl. Proin non mi sed ligula mattis consectetur. Sed sagittis nisl mollis
      felis sollicitudin lobortis. Praesent rhoncus turpis id viverra gravida.
      Pellentesque consequat turpis ut ex commodo, in volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel cursus non, semper quis lectus. Fusce at
      justo vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
      mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>
"
`;

exports[`end-to-end integration prints formatted output to specified width for --indent 1`] = `
"<!DOCTYPE html>
<html>
    <head>
        <meta charset=\\"utf-8\\" />
        <title>Sample Web Page</title>
    </head>
    <body>
        <h1>Sample Web Page</h1>
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean
            aliquet molestie vestibulum. Nunc malesuada lobortis mi et commodo.
            Aenean at diam non lectus porta laoreet. Curabitur in aliquam augue.
            Sed in maximus ligula. Suspendisse iaculis tincidunt varius. In sit
            amet nulla in erat consequat tempor. Praesent eget urna ornare,
            mattis magna et, interdum nisl. Proin non mi sed ligula mattis
            consectetur. Sed sagittis nisl mollis felis sollicitudin lobortis.
            Praesent rhoncus turpis id viverra gravida. Pellentesque consequat
            turpis ut ex commodo, in volutpat ligula tristique. Vestibulum mi
            nisl, accumsan vel cursus non, semper quis lectus. Fusce at justo
            vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
            mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et
            varius lectus rhoncus eget.
        </p>
    </body>
</html>
"
`;

exports[`end-to-end integration prints formatted output to specified width for --width 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet,
      consectetur adipiscing elit.
      Aenean aliquet molestie
      vestibulum. Nunc malesuada
      lobortis mi et commodo. Aenean at
      diam non lectus porta laoreet.
      Curabitur in aliquam augue. Sed in
      maximus ligula. Suspendisse
      iaculis tincidunt varius. In sit
      amet nulla in erat consequat
      tempor. Praesent eget urna ornare,
      mattis magna et, interdum nisl.
      Proin non mi sed ligula mattis
      consectetur. Sed sagittis nisl
      mollis felis sollicitudin
      lobortis. Praesent rhoncus turpis
      id viverra gravida. Pellentesque
      consequat turpis ut ex commodo, in
      volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel
      cursus non, semper quis lectus.
      Fusce at justo vitae justo
      eleifend auctor. Sed nisi diam,
      dapibus interdum luctus mattis,
      venenatis sit amet orci. Vivamus
      tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>
"
`;

exports[`end-to-end integration prints formatted output to specified width for -i 1`] = `
"<!DOCTYPE html>
<html>
    <head>
        <meta charset=\\"utf-8\\" />
        <title>Sample Web Page</title>
    </head>
    <body>
        <h1>Sample Web Page</h1>
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean
            aliquet molestie vestibulum. Nunc malesuada lobortis mi et commodo.
            Aenean at diam non lectus porta laoreet. Curabitur in aliquam augue.
            Sed in maximus ligula. Suspendisse iaculis tincidunt varius. In sit
            amet nulla in erat consequat tempor. Praesent eget urna ornare,
            mattis magna et, interdum nisl. Proin non mi sed ligula mattis
            consectetur. Sed sagittis nisl mollis felis sollicitudin lobortis.
            Praesent rhoncus turpis id viverra gravida. Pellentesque consequat
            turpis ut ex commodo, in volutpat ligula tristique. Vestibulum mi
            nisl, accumsan vel cursus non, semper quis lectus. Fusce at justo
            vitae justo eleifend auctor. Sed nisi diam, dapibus interdum luctus
            mattis, venenatis sit amet orci. Vivamus tempus tellus nibh, et
            varius lectus rhoncus eget.
        </p>
    </body>
</html>
"
`;

exports[`end-to-end integration prints formatted output to specified width for -w 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Sample Web Page</title>
  </head>
  <body>
    <h1>Sample Web Page</h1>
    <p>
      Lorem ipsum dolor sit amet,
      consectetur adipiscing elit.
      Aenean aliquet molestie
      vestibulum. Nunc malesuada
      lobortis mi et commodo. Aenean at
      diam non lectus porta laoreet.
      Curabitur in aliquam augue. Sed in
      maximus ligula. Suspendisse
      iaculis tincidunt varius. In sit
      amet nulla in erat consequat
      tempor. Praesent eget urna ornare,
      mattis magna et, interdum nisl.
      Proin non mi sed ligula mattis
      consectetur. Sed sagittis nisl
      mollis felis sollicitudin
      lobortis. Praesent rhoncus turpis
      id viverra gravida. Pellentesque
      consequat turpis ut ex commodo, in
      volutpat ligula tristique.
      Vestibulum mi nisl, accumsan vel
      cursus non, semper quis lectus.
      Fusce at justo vitae justo
      eleifend auctor. Sed nisi diam,
      dapibus interdum luctus mattis,
      venenatis sit amet orci. Vivamus
      tempus tellus nibh, et varius
      lectus rhoncus eget.
    </p>
  </body>
</html>
"
`;

exports[`end-to-end integration prints help message when --help given 1`] = `
"purl <args> uri [...uri]

Options:
      --version  Show version number                                   [boolean]
  -w, --width    max width of final output                [number] [default: 80]
  -i, --indent   indent width for each line                [number] [default: 2]
      --help     Show help                                             [boolean]"
`;
